handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -809,6 +828,11 @@ public final SSLEngineResult unwrap(
                 throw new NotSslRecordException("not an SSL/TLS record");
             }
 
+            // IMPORTANT:
+            //
+            // This calculation only works as we do not support compression and set
```

Comment:
This calculation only works as we do not support compression. We explicitly disable compression by setting the SSL_OP_NO_COMPRESSION option. If compression were enabled it is possible that the plain text length is > the packet length.